handout/routing/child_routes.md

```diff
@@ -1,60 +1,92 @@
 
-Here's an example:
+Where would the components for these child routes be displayed? Just like we had a `<router-outlet></router-outlet>` for the root application component, we would have a router outlet inside the `ProductDetails` component. The components corresponding to the child routes of `product-details` would be placed in the router outlet in `ProductDetails`.
 
 ```javascript
+import { Component } from '@angular/core';
+import { ROUTER_DIRECTIVES, ActivatedRoute } from '@angular/router';
+
 @Component({
-  directives: [ROUTER_DIRECTIVES]
-  selector: 'component-one-container',
-  template: `Component One Container
-  <br/>
-  <div style="border: 1px solid red">
+  selector: 'product-details',
+  template: `
+    <p>Product Details: {{id}}</p>
+    <!-- Product information -->
+    <nav>
+      <a [routerLink]="['overview']">Overview</a>
+      <a [routerLink]="['specs']">Technical Specs</a>
+    </nav>
     <router-outlet></router-outlet>
-  </div>
-  `
+    <!-- Overview & Specs components get added here by the router -->
+  `,
+  directives: [ROUTER_DIRECTIVES]
 })
-@RouteConfig([{
-    path: '/',
-    component: ComponentOne,
-    as: 'ComponentOne',
-    useAsDefault: true
-  }, {
-    path: '/component-three-nested/:message',
-    component: ComponentThree,
-    as: 'ComponentThree'
+export default class ProductDetails {
+  private id;
+
+  constructor(private route: ActivatedRoute) {}
+
+  private ngOnInit() {
+    this.sub = this.route.params.subscribe(params => {
+       this.id = +params['id']; // (+) converts string 'id' to a number
+    });
   }
-, {
-  path: '/component-one-child-one',
-  component: ComponentOneChildOne,
-  as 'ComponentOneChildOne'
-}, {
-  path: '/component-one-child-two',
-  component: ComponentOneChildTwo,
-  as 'ComponentOneChildTwo'
-}, {
-  path: '/component-one-child-three/:message',
-  component: ComponentThree,
-  as 'ComponentThree'
-}])
-export default class ComponentOneContainer {
 
+  private ngOnDestroy() {
+    this.sub.unsubscribe();
+  }
 }
+```
+
+Alternatively, we could also specify the route such that the `overview` route URL simply be:
+
+`localhost:3000/product-details/3`
 
+```javascript
+export const routes: RouterConfig = [
+  { path: '', redirectTo: 'product-list', pathMatch: 'full' },
+  { path: 'product-list', component: ProductList },
+  { path: 'product-details/:id', component: ProductDetails,
+    children: [
+      { path: '', component: Overview },
+      { path: 'specs', component: Specs }
+    ]
+  }
+];
 ```
-[View Example](https://plnkr.co/edit/qCYF138UXe6l9CiTXggG?p=preview)
+Since the `Overview` child route of `product-details` has an empty path, it will be loaded by default. The `specs` child route remains the same.
+
+[View Example with child routes](https://plnkr.co/edit/QlMe6pMINxJGTdA3xm0B?p=preview)
+
+[View Example with route params & child routes](https://plnkr.co/edit/g88pvg4LQq7XVJo7iG9b?p=preview)
```

Comment:
Was playing around with an example to show the difference between

```html
<a [routerLink]="route"></a>
<a [routerLink]="/route"></a>
<a [routerLink]="../route"></a>
```

Might be worth trying to incorporate the lookup of 'current vs parent vs root' routes